debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -74,7 +77,7 @@ public R apply(R r) {
         String op = debeziumEventValue.getString(Envelope.FieldName.OPERATION);
 
         // Skipping deletes
-        if (op.equals(Envelope.Operation.DELETE.code())) {
+        if (op.equals(Envelope.Operation.DELETE.code()) && !routeTombstoneOnEmptyPayload) {
```

Comment:
I don't think DELETEs should be handled. Otherwise e.g. the deletion of an "OrderCreated" event would trigger th tombstone potentially. The empty/null payload should be the sole trigger for this.

Reply:
:doh: yes that is a good point!